app/assets/javascripts/discourse/app/pre-initializers/sniff-capabilities.js

```diff
@@ -1,4 +1,18 @@
 // Initializes an object that lets us know about browser's capabilities
+
+const APPLE_NAVIGATOR_PLATFORMS = [
+  "iPhone",
+  "iPod",
+  "iPad",
+  "iPhone Simulator",
+  "iPod Simulator",
+  "iPad Simulator",
+  "Macintosh",
+  "MacIntel",
+];
```

Comment:
Note that I omitted some devices like:

"MacPPC"
"Mac68K"
"Pike v7.6 release 92" (opera)
"Pike v7.8 release 517" (opera)

as we support the [latest version of major browsers](https://www.discourse.org/about#browser). 😸

Reply:
Could we simplify this to just `iPhone/iPod/iPad/Mac`? 

Also, I added something similar recently: 
https://github.com/discourse/discourse/blob/main/app/assets/javascripts/discourse/app/lib/utilities.js#L481 
it would be nice to centralize everything, that `utilities` library probably doesn't have the capabilities context.

Reply:
I actually found that `helperContext().capabilities` is usable in utilities, and we can get capabilities! https://github.com/discourse/discourse/blob/main/app/assets/javascripts/discourse/app/lib/utilities.js#L304

The tiny problem now is that utilities has the same function as capabilities (see isApple in both). Being not a fan of generic utility functions that can have everything, I think I favour having device specific checks having `sniff-capabilities` and use them from utilities. 

However looking at how utiliites.isAppleDevice is used, I don't think I will consolidate the definitions yet. https://github.com/discourse/discourse/blob/main/app/assets/javascripts/discourse/app/lib/media-optimization-utils.js shows a comment on browser specific features, but did not mention what would happen if `isAppleDevice` is using Chrome.